src/dvc_objects/fs/generic.py

```diff
@@ -115,40 +271,75 @@ def _try_links(
+    batch_size: Optional[int] = None,
+    on_error: Optional[TransferErrorHandler] = None,
 ) -> None:
-    try:
-        assert not (hardlink and links)
-        if hardlink:
-            links = links or ["reflink", "hardlink", "copy"]
-        else:
-            links = links or ["reflink", "copy"]
-
-        _try_links(
-            links, from_fs, from_path, to_fs, to_path, callback=callback
-        )
-    except OSError as exc:
-        # If the target file already exists, we are going to simply
-        # ignore the exception (#4992).
-        #
-        # On Windows, it is not always guaranteed that you'll get
-        # FileExistsError (it might be PermissionError or a bare OSError)
-        # but all of those exceptions raised from the original
-        # FileExistsError so we have a separate check for that.
-        if isinstance(exc, FileExistsError) or (
-            os.name == "nt"
-            and exc.__context__
-            and isinstance(exc.__context__, FileExistsError)
-        ):
-            logger.debug("'%s' file already exists, skipping", to_path)
-            return None
-
-        raise
+    if isinstance(from_path, str):
+        from_path = [from_path]
+    if isinstance(to_path, str):
+        to_path = [to_path]
+    assert len(from_path) == len(to_path)
+
+    assert not (hardlink and links)
+    if hardlink:
+        links = links or ["reflink", "hardlink", "copy"]
+    else:
+        links = links or ["reflink", "copy"]
+
+    callback.set_size(len(from_path))
+    # Try to link files sequentially. If/when the only remaining link type is
+    # copy, the remaining copy operations will be batched.
+    for i, (from_p, to_p) in enumerate(zip(from_path, to_path)):
+        if links[0] == "copy":
+            return copy(
+                from_fs,
+                from_path[i:],
+                to_fs,
+                to_path[i:],
+                callback=callback,
+                batch_size=batch_size,
+                on_error=on_error,
+            )
+        try:
+            _try_links(
+                links,
+                from_fs,
+                from_p,
+                to_fs,
+                to_p,
+                callback=callback,
+            )
+        except OSError as exc:
+            # If the target file already exists, we are going to simply
+            # ignore the exception (#4992).
+            #
+            # On Windows, it is not always guaranteed that you'll get
+            # FileExistsError (it might be PermissionError or a bare OSError)
+            # but all of those exceptions raised from the original
+            # FileExistsError so we have a separate check for that.
+            if isinstance(exc, FileExistsError) or (
+                os.name == "nt"
+                and exc.__context__
+                and isinstance(exc.__context__, FileExistsError)
+            ):
+                logger.debug("'%s' file already exists, skipping", to_path)
+                return None
```

Comment:
Should be `continue` instead. Same above in `copy`. Looks like this was an oversight when making this work with multiple files.